### DISCUSSION TITLE
Process component - child outliving parent on one environment, but not another

### DISCUSSION THREAD

**Discussion by JKingsnorth** [symfony/symfony#48113] [Q&A]
Hi community!

We're using https://symfony.com/doc/current/components/process.html to start a server process from a PHP script, but then immediately return a 200 response, leaving the server process running.
```
$process->start();
return Response...
```

On our local dev environment, the child process continues after the parent script has returned a response. 👍 

On another environment, the child process is killed when the parent script returns a response. 👎 

What configuration might be causing this difference in behavior?

Both are Linux/Apache/PHP8.1 environments; although we know they're not identical. I'm looking for some clues about where to look.

Thanks very much.

**Comment by MatTheCat** [comment]
From https://symfony.com/doc/current/components/process.html#running-processes-asynchronously

> If a `Response` is sent **before** a child process had a chance to complete, the server process will be killed (depending on your OS).

Are OSes the same?

If your process is short-lived you should probably run it synchronously. Else,

> use a job queue instead

**Reply by JKingsnorth** [reply]
Thanks @MatTheCat .

The local environment it is 'working' on is Debian. The environment where the child process is killed is Red Hat EL.

I also saw that in the documentation, but I would be interested to know more about where/why exactly this is configured, if it is a difference in Linux operating systems.

I agree that using a job queue would be a more standard approach to this problem in a real scenario, thanks for the suggestion. But I am still interested to know more about this example, and what is causing the different behaviour.

**Reply by MatTheCat** [reply]
Are these two OSes running bare metal or in a container? If so, does an init process run in it?

**Reply by StrayCudar** [reply]
Thanks, I'm working with @JKingsnorth.. The local Debian environment is running in a lando docker set up, no init process running. The RHEL server is running in VMware with systemd.
